libs/fluid-elements/tab-group/src/lib/tab/tab.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  LitElement,
+  CSSResult,
+  property,
+  TemplateResult,
+  html,
+} from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map';
+import styles from './tab.scss';
+
+let _unique = 0;
+
+/** Custom event implementation that fires when a tab is clicked providing the active tab id  */
+export class FluidTabActivatedEvent extends CustomEvent<any> {
+  constructor(public activeTab: string) {
+    super('tabActivated', { bubbles: true, composed: true });
+  }
+}
+
+export class FluidTabDisabledEvent extends CustomEvent<any> {
+  constructor(public disableTab: string) {
+    super('disabled', { bubbles: true, composed: true });
+  }
+}
+
+/**
+ * This is a experimental version of the tab component
+ * It registers itself as `fluid-tab` custom element.
+ */
+export class FluidTab extends LitElement {
+  /** Styles for the tab component */
+  static get styles(): CSSResult {
+    return styles;
+  }
+
+  /**
+   * Defines the tab element with an id attribute
+   * @attr
+   * @type string
+   */
+  @property({ type: String, reflect: true })
+  tabid = `fluid-tab-${_unique++}`;
+
+  /**
+   * Defines whether a tab is disabled or not
+   * @attr
+   * @type boolean
+   */
+  @property({ type: Boolean, reflect: true })
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    const oldValue = this._disabled;
+    if (this._disabled !== value) {
+      this._disabled = value;
+      this.requestUpdate('disabled', oldValue);
+      if (this._disabled) {
+        this.active = false;
+        this.tabindex = -1;
+        this.dispatchEvent(new FluidTabDisabledEvent(this.tabid));
+      }
+    }
+  }
+  private _disabled = false;
+
+  /**
+   * Defines the tabindex attribute
+   * @attr
+   * @type number
+   */
+  @property({ type: Number, reflect: true })
+  get tabindex(): number {
+    return this._tabindex;
+  }
+  set tabindex(value: number) {
+    this._tabindex = value;
+    this.requestUpdate('tabindex');
```

Comment:
Please add a `//TODO: ` to figure out why the requestUpdate with the old value does not work

Reply:
here aswell